src/storage-client/src/controller.rs

```diff
@@ -2108,62 +2108,14 @@ mod persist_write_handles {
 
                                             futs.push(async move {
                                                 let persist_upper = persist_upper.clone();
-                                                let mut result =
                                                 write
                                                     .compare_and_append(
                                                         updates.clone(),
                                                         persist_upper.clone(),
                                                         new_upper.clone(),
                                                     )
                                                     .instrument(span.clone())
-                                                    .await;
-
-                                                // Indeterminate results can occur when persist is not certain
```

Comment:
I'm so happy about this specific block of code going away! 😍 

Specifically because of
```
// If the upper frontier is the new frontier, then because
// of mutual exclusion of writes, no other writer should be
// advancing the frontier to `new_upper`.
//
// TODO: This may succeed if `new_upper` is where we cut over
// to a new leader, who advanced tables to `new_upper` when it
// started. In that case, a success here will soon be followed
// by a failure on our next interaction with the catalog stash,
// but we would incorrectly think this committed and may serve
// results in the meantime.
```